docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,713 @@
+# PVXS Architecture
+
+This document provides an overview of the PVXS architecture, design principles, and internal structure.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Design Principles](#design-principles)
+- [Architecture Components](#architecture-components)
+- [Data Layer](#data-layer)
+- [Network Layer](#network-layer)
+- [Client Architecture](#client-architecture)
+- [Server Architecture](#server-architecture)
+- [IOC Integration](#ioc-integration)
+- [Threading Model](#threading-model)
+- [Memory Management](#memory-management)
+- [Protocol Implementation](#protocol-implementation)
+- [Comparison with Other EPICS Modules](#comparison-with-other-epics-modules)
+
+## Overview
+
+PVXS is a modern C++ library implementing the PVAccess (PVA) protocol for EPICS. It provides three main layers:
+
+1. **Data Layer** - Type-safe data containers (`pvxs::Value`)
+2. **Network Layer** - PVAccess protocol implementation
+3. **Application Layer** - Client and Server APIs
+
+```
+┌─────────────────────────────────────────────────────────────┐
+│                    Application Layer                        │
+│  ┌──────────────┐              ┌──────────────┐            │
```

Comment:
fyi. an alternative to ascii art, [sphinx](https://www.sphinx-doc.org/en/master/usage/extensions/graphviz.html) can process [graphviz](https://graphviz.org/) diagrams.  An example from [`p4p.gw`](https://github.com/epics-base/p4p/blob/77e04080162d0be025a22a9e2e5fcc7694da0b69/documentation/gw.rst?plain=1#L38).